### PR TITLE
Util+Knife Stats & Rename

### PR DESCRIPTION
**Pull request by PhlexPlexico**
These changes include these new values if they are ever included.

### CODE REVIEW

**Comment by PhlexPlexico** [comment]
https://github.com/splewis/get5/pull/612 as reference.

**Comment by PhlexPlexico** [comment]
@nickdnk to avoid clogging up your PR on get5 with some questions I'd thought it'd be best to ask here.

I believe the only added change for stats (to record/send/add to db per latest pushes) would now just be friendlies flashed? I see that the flash assist is now using a bool, but it looks like it's still sending either a 1 or 0, so it shouldn't matter incrementing.

As for the friendlies flashed, I really like that as a stat, can go to show how effective your flashes are on enemies vs. friendlies, neat!

**Comment by nickdnk** [comment]
Hello @PhlexPlexico

1. For the database, we did not have `enemies_flashed` before either. I added both `enemies_flashed` and `friendlies_flashed` in my latest push. There was no tracking of flashes, only flash assists. I also re-worked the flash-assist logic to use the game-provided events instead of the custom "who threw which flashbang at what time"-workarond that was implemented before.
2. I'm not sure I understand what you mean by "but it looks like it's still sending either a 1 or 0, so it shouldn't matter incrementing." - can you elaborate? I don't believe I am incrementing a boolean anywhere. For the logging system we're now providing a user object (the assister) and a boolean indicating whether the assist was a flash-assist or a regular assist, as the game does not allow you to get both a regular assist and a flash-assist at the same time. Flash-assists are only triggered if you flashed an enemy and *did not* deal enough damage for a regular assist, in which case your flash-assist would be overriden.

File: scripting/include/get5.inc [hunk 1]
```diff
@@ -174,6 +174,8 @@ native int Get5_IncreasePlayerStat(int client, const char[] statName, int amount
 #define STAT_TEAMKILLS "teamkills"
 #define STAT_SUICIDES "suicides"
 #define STAT_DAMAGE "damage"
+#define STAT_UTILITY_DAMAGE "util_damage"
+#define STAT_KNIFE_KILLS "knife_kills"
 #define STAT_HEADSHOT_KILLS "headshot_kills"
```
Comment (nickdnk) [inline scripting/include/get5.inc:179]: @PhlexPlexico: As you can see, both `enemies_flashed` and `friendlies_flashed` are missing here.

**Comment by PhlexPlexico** [comment]
Ahh okay, as for point 1 I see, thanks! It's hard to view some of these changes on mobile, so forgive me haha. 

As for point 2, I see that makes more sense. From what I could read in the changes to the event logger was checking that assist boolean. Hence why I wanted to clarify with a stupid question because it's hard for me to read this late at night haha. Thanks for the thorough explanation and once I get back at it, I'll add in the new includes and make sure they get sent in to the API :)

Thank you!!

**Comment by nickdnk** [comment]
You're welcome. Now we just need splewis to accept the PR. I run with my changes in "production" and it seems everything is good.

**Comment by PhlexPlexico** [comment]
Closes #18 and #15

**Comment by PhlexPlexico** [comment]
Closes #20

**Comment by stranden** [comment]
@PhlexPlexico Will this get merged into the master at some point? And with a new release available? I really like your work. Thanks a lot - also to you @nickdnk! 👍

**Comment by PhlexPlexico** [comment]
Hey @stranden! Thanks for the kind words, however I'm still at the mercy of whenever these changes get merged into get5 itself. I'm not exactly sure how nicely this would play with different `get5,inc` files after compilation. If that gets merged, then I will be able to merge this, plus the API and front-end, but until then I have to wait.

Thanks!

**Comment by nickdnk** [comment]
@stranden You're gonna have to bump https://github.com/splewis/get5/pull/612 if you want this released.

**Pull request merged**